### PR TITLE
dotcover support

### PR DESCRIPTION
**Pull request by mholo65**
Still needs some work. Wanted to check if I'm on the right path here before doing more work. Plan is to also add support for other [commands](https://www.jetbrains.com/dotcover/help/dotCover__Console_Runner_Commands.html) than analyse, e.g. cover, merge, report and delete 

Fixes issue #658

### CODE REVIEW

**Comment by patriksvensson** [comment]
@mholo65 I think this looks great :+1: 

I think it would suffice with the Analyze command for this PR and add the others after that as separate PR:s to keep them focused.

What do you think?

**Comment by mholo65** [comment]
Thanks, sounds good to me! I'll finalize this by adding support for more options and adding more unit tests. Expect an updated PR by the end of the week.

**Comment by patriksvensson** [comment]
@mholo65 Sounds good to me :+1:

**Comment by mholo65** [comment]
@patriksvensson I think this may be done for now, what do you think?

**Comment by patriksvensson** [comment]
@mholo65 Will review it ASAP.

File: src/Cake.Common/Tools/DotCover/Analyse/DotCoverAnalyser.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.DotCover.Analyse
+{
+    /// <summary>
+    /// DotCover Analyser builder.
+    /// </summary>
+    public class DotCoverAnalyser : DotCoverTool<DotCoverAnalyseSettings>
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DotCover/Analyse/DotCoverAnalyser.cs]: Should be sealed.

File: src/Cake.Common/Tools/DotCover/Analyse/DotCoverAnalyseSettings.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+using Cake.Common.Tools.DotCover;
+using Cake.Core.IO;
+using Cake.Core.Tooling;
+
+namespace Cake.Common.Tools.DotCover.Analyse
+{
+    /// <summary>
+    /// Contains settings used by <see cref="DotCoverAnalyser" />.
+    /// </summary>
+    public class DotCoverAnalyseSettings : ToolSettings
```
Comment (patriksvensson) [inline src/Cake.Common/Tools/DotCover/Analyse/DotCoverAnalyseSettings.cs]: Should be sealed.

**Comment by patriksvensson** [comment]
@mholo65 This looks good to me. Left some minor comments but otherwise this looks solid to me. Great job! :+1:

**Comment by mholo65** [comment]
Thanks @patriksvensson, should be fixed now. Also fixed documentation on filters for alias DotCoverAnalyse since it was incorrect.

**Comment by patriksvensson** [comment]
@mholo65 Merged!
Thank you for this! Awesome work! :+1:

**Pull request merged**